programs/relayer/Dockerfile

```diff
@@ -0,0 +1,14 @@
+ARG SP1_VERSION="v4.1.0"
```

Comment:
The arg can be retrieved from `Cargo.toml`. We don't need to add this to `docker.yml` yet but would be a nice improvement. I used to do this in our workflows.

Reply:
doesnt seem like a super easy thing to do, so i'll add a todo in the file

Reply:
not easy due to us using either 4.1 in the cargo.toml or 4.1.1 in the Cargo.lockfile, and neither of these versions correspond to a built image (4.1.0 is latest)